packages/typespec-vscode/src/code-action-provider.ts

```diff
@@ -47,9 +49,97 @@ export class TypeSpecCodeActionProvider implements vscode.CodeActionProvider {
         );
       }
     });
+
+    actions.push(...(await this.createInstallPackageByNpm(context, actions)));
+
     return actions;
   }
 
+  private async createInstallPackageByNpm(
+    context: vscode.CodeActionContext,
+    actions: vscode.CodeAction[],
+  ): Promise<vscode.CodeAction[]> {
+    // When the corresponding node dependency package is not installed,
+    // consider generating a quick fix to install via npm command
+    const diagnostics = context.diagnostics.filter(
+      (diagnostic) =>
+        diagnostic.source === "TypeSpec" &&
+        diagnostic.code &&
+        diagnostic.code === "import-not-found",
+    );
+    if (diagnostics.length === 0) {
+      return [];
+    }
+
+    for (const diagnostic of diagnostics) {
+      if (
+        "data" in diagnostic &&
+        diagnostic.data &&
+        typeof diagnostic.data === "object" &&
+        "file" in diagnostic.data
+      ) {
+        // The message content is `Couldn't resolve import "@typespec/http"`.
+        // The compiler's diagnostics do not provide a specific attribute to obtain the package name,
+        // so a regular expression is used to extract the package name within the double quotes
+        const targetPackage = diagnostic.message.replace(/.*"([^"]+)".*/, "$1");
+        const packageFile = diagnostic.data.file as string;
+        const packageFileUri = vscode.Uri.parse(packageFile);
+        const { packageJsonFolder, packageJson } = await searchAndLoadPackageJson(
+          getDirectoryPath(packageFileUri.fsPath),
+        );
+
+        let packageName: string = "";
+        let projectFolder: string | undefined;
+        if (packageJsonFolder === undefined) {
+          // If the package.json file is not found, install the specified `targetPackage` directly
+          packageName = targetPackage;
+          projectFolder = vscode.workspace.workspaceFolders?.[0].uri.fsPath;
+        } else if (
+          !targetPackage.startsWith("./") &&
+          !isPathAbsolute(targetPackage) &&
+          packageJson &&
+          ((packageJson.peerDependencies && targetPackage in packageJson.peerDependencies) ||
+            (packageJson.dependencies && targetPackage in packageJson.dependencies) ||
+            (packageJson.devDependencies && targetPackage in packageJson.devDependencies))
+        ) {
+          // If the package.json file is found, and the file contains the specified package,
+          // it is installed according to the package name in the package.json file.
+          projectFolder = packageJsonFolder;
+        }
+
+        if (projectFolder) {
+          const action = this.createInstallPackageCodeAction(
+            diagnostic,
+            projectFolder,
+            packageName,
+          );
+          if (!actions.some((item) => item.title === action.title)) {
```

Comment:
let the caller to make the call to include the action or not, the function should just generate the action